eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -86,12 +86,14 @@ jobs:
           Exit 1
         }
 
-        $sdkRepoNameForSpecGen = $sdkRepoName
-        if ($sdkRepoName.EndsWith('-for-net')) {
-          $sdkRepoNameForSpecGen = $sdkRepoName + "-track2"
+        if ($env:BUILD_REASON -eq 'PullRequest') {
```

Comment:
Can we default the SpecRepoCommit parameter to this variable, so we don't have to override here in code?

Reply:
Made this change and changed the value in entry yml too.
This is a [testing commit](https://github.com/azure-sdk/azure-sdk-for-java/commit/4a2db7c74eaa8b67f7fed79a39a9726ec3ea7261#diff-e0126afe55b455125a857b72ca29be278c8628ae725833622cbc5b8cc1f6e708) generated.